### ISSUE TITLE
Fix secret registry plugin integration tests for tanzu framework

### ISSUE THREAD

**Issue by shivaani0505** [vmware-tanzu/tanzu-framework#1141]
Some of the package plugin integration tests are disabled. Need to fix the testcases and bring them up.
https://github.com/vmware-tanzu/tanzu-framework/blob/d05775ced0163170f84f45206414a7be2f344eff/pkg/v1/tkg/test/tkgpackageclient/package_plugin_integration_test.go#L388-L440

**Comment by blc1996** [comment]
The tests are for testing private package repository aceess with package plugin. The tests should fail to access private package repository without access secret, but it was successful in the pipeline.

**Comment by blc1996** [comment]
After some investigations, it turns out the auth feature for `projects-stg.registry.vmware.com/tkg` is not working because of some CDN cache issue.
The AI is to switch to a gcr private registry for testing.

**Comment by blc1996** [comment]
We have the following 3 options to enable the tests again. Need inputs from team members.
* Use another registry like GCR, but it is not best practice to upload credentials to a publicly viewable Girhub repo. Also, I neeed to fix this issue before I could do it (GRC credentials has white space in it, and it’s not supported by our test lib).
* Spin up a local registry in cluster. I tried to do it but found an issue with hostNetwork option enabled by kapp-controller. Pods using host network will not use cluster DNS, instead it will use an external DNS as specified in it’s /etc/resolv.conf. We will have following error when resolve the registry
```
Fetching image:
        Get "https://registry-svc.registry.svc.cluster.local:443/v2/": dial tcp: lookup registry-svc.registry.svc.cluster.local on 10.162.204.1:53: no such host; Get "http://registry-svc.registry.svc.cluster.local:443/v2/": dial tcp: lookup registry-svc.registry.svc.cluster.local on 10.162.204.1:53: no such host
```
Setting
```
hostNetwork: true
dnsPolicy: ClusterFirstWithHostNet
```
would solve the problem, but is it a good idea to change the deployment spec during an integration test? Should we expose dbsPolicy as a configurable param?
* Create our own private registry and use that instead?
